src/Microsoft.Health.Core/Features/Events/EventGridSink.cs

```diff
@@ -0,0 +1,72 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Messaging.EventGrid;
+using EnsureThat;
+using Microsoft.Health.Abstractions.Data;
+using Microsoft.Health.Abstractions.Features.Events;
+
+namespace Microsoft.Health.Core.Features.Events
+{
+    /// <summary>
+    /// EventGridSink.
+    /// </summary>
+    /// <typeparam name="T">T of Type IEvent.</typeparam>
+    public class EventGridSink<T> : ISink<T>
+        where T : class, IEvent
+    {
+        private readonly IEventGridPublisher _client;
```

Comment:
Nit: should the variable name be _eventGridPublisher ?

Reply:
Yes, I should have renamed this also after refactor. Shall do.

Reply:
fixed.